test/runtests.jl

```diff
@@ -140,9 +140,13 @@ end
         @test mean(x) == sum(x) / length(x)
     end
     @test mean(Number[1, 1.5, 2+3im]) === 1.5+1im # mixed-type array
-    @test (@inferred mean(Int[])) === 0.0/0
-    @test (@inferred mean(Float64[])) === 0.0/0
-    @test (@inferred mean(Float32[])) === 0.0f0/0
+    @test mean(v for v in Number[1, 1.5, 2+3im]) === 1.5+1im
+    @test (@inferred mean(Int[])) === 0/0
+    @test (@inferred mean(Float32[])) === 0.f0/0    
+    @test (@inferred mean(Float64[])) === 0/0
+    @test (@inferred mean(filter(x -> true, Int[]))) === 0/0
+    @test (@inferred mean(filter(x -> true, Float32[]))) === 0.f0/0
+    @test (@inferred mean(filter(x -> true, Float64[]))) === 0/0
```

Comment:
`filter` returns a vector, you need `Iterators.filter`. Probably also `using Iterators` at the top of the file.
```suggestion
    @test (@inferred mean(Iterators.filter(x -> true, Int[]))) === 0/0
    @test (@inferred mean(Iterators.filter(x -> true, Float32[]))) === 0.f0/0
    @test (@inferred mean(Iterators.filter(x -> true, Float64[]))) === 0/0
```